Building metatensor-torch on izar fails with "Unknown CUDA Architecture Name 9.0a in CUDA_SELECT_NVCC_ARCH_FLAGS"

Heya, I've just built `metatensor-torch` with `pip install metatensor-torch --no-binary=metatensor-torch` on `izar`.

With the following environment:

```
module purge
module load git
module load git-lfs
module load gcc/11.3.0
module load python/3.10.4
module load cuda/12.1.1
module load cudnn/8.9.7.29-12

# ... python env

pip install metatensor-torch --no-binary=metatensor-torch
```

This fails with

```
...
      -- USE_CUDNN is set to 0. Compiling without cuDNN support
      -- USE_CUSPARSELT is set to 0. Compiling without cuSPARSELt support
      -- Automatic GPU detection failed. Building for common architectures.
      -- Autodetected CUDA architecture(s): 5.0;8.0;8.6;8.9;9.0;9.0a
      CMake Error at /tmp/pip-build-env-ldt33l39/normal/lib/python3.10/site-packages/torch/share/cmake/Caffe2/Modules_CUDA_fix/upstream/FindCUDA/select_compute_arch.cmake:227 (message):
        Unknown CUDA Architecture Name 9.0a in CUDA_SELECT_NVCC_ARCH_FLAGS

...
```

This appears to be [known](https://github.com/pytorch/pytorch/issues/113948) and can be fixed with 

```
export TORCH_CUDA_ARCH_LIST="8.0 8.6 8.9 9.0"
```

Maybe we should add this to the installation docs?

--- a/docs/src/installation.rst
+++ b/docs/src/installation.rst
@@ -183,12 +183,17 @@ and use it will depend on the programming language you are using.
 
         .. code-block:: bash
 
-            pip install --extra-index-url https://download.pytorch.org/whl/cpu metatensor-torch  --no-binary=metatensor-torch
+            pip install --extra-index-url https://download.pytorch.org/whl/cpu metatensor-torch --no-binary=metatensor-torch
 
         A similar index URL can be used to install the ROCm (AMD GPU) version of
         PyTorch, please refer to the `corresponding documentation
         <https://pytorch.org/get-started/locally/>`_.
 
+        .. seealso::
+
+            Some potential build failures and corresponding workarounds are
+            listed at the end of the :ref:`install-torch-cxx` installation
+            instructions.
 
 
     .. tab-item:: TorchScript C++
@@ -250,6 +255,34 @@ and use it will depend on the programming language you are using.
         |                                      | search for dependencies. This list should     |                |
         |                                      | include the path to metatensor and torch      |                |
         +--------------------------------------+-----------------------------------------------+----------------+
+
+        **Workaround for some build errors**
+
+        The CMake configuration used by libtorch sometimes fails to setup the
+        build environment. You'll find here a list of some known build failures
+        and how to workaround them.
+
+        - .. code-block:: text
+
+              Unknown CUDA Architecture Name 9.0a in CUDA_SELECT_NVCC_ARCH_FLAGS
+
+          This can happen when building with a CUDA-enabled version of torch and
+          a recent version of cmake. This issue is tracked at
+          https://github.com/pytorch/pytorch/issues/113948. To work around it,
+          you can ``export TORCH_CUDA_ARCH_LIST="8.0 8.6 8.9 9.0"`` in your
+          environment before building the code.
+
+        - .. code-block:: text
+
+              Imported target "torch" includes non-existent path
+                [...]/MKL_INCLUDE_DIR-NOTFOUND"
+              in its INTERFACE_INCLUDE_DIRECTORIES.
+
+          This can happen when building for x86_64 Linux when MKL is not
+          available on the current machine. Since MKL is a completely optional
+          dependency, you can silence the error by running cmake with the
+          ``-DMKL_INCLUDE_DIR=/usr/include`` option.
+
 
 
 Installing a development version

